### PR TITLE
client,bridgev2: Add support for MSC4190

### PR DESCRIPTION
**Pull request by onestacked**
Adds [MSC4190](https://github.com/matrix-org/matrix-spec-proposals/pull/4190) support.

### CODE REVIEW

**Review by tulir — COMMENTED** [review]
The config option goes in `bridgev2/bridgeconfig` + `bridgev2/matrix/mxmain/example-config.yaml` (no need to update the legacy `bridge` module)

`SetAppServiceDeviceID` probably also has to be set in the e2ee client

**Comment by onestacked** [comment]
I've now added the config flag in bridgev2 and removed some changes not needed for bridgev2.

I've tested this with element-hq/synapse#17705 and [MSC3861](https://github.com/matrix-org/matrix-spec-proposals/pull/3861).
Requires msc3202_device_masquerading: true to be set in the synapse config.

**Review by tulir — APPROVED** [review]

File: bridgev2/bridgeconfig/legacymigrate.go [hunk 1]
```diff
@@ -63,6 +63,7 @@ func doMigrateLegacy(helper up.Helper, python bool) {
 	}
 	helper.Copy(up.Bool, "appservice", "ephemeral_events")
 	helper.Copy(up.Bool, "appservice", "async_transactions")
+	helper.Copy(up.Bool, "appservice", "msc4190")
```
[the remark this answers is not in the document]
Comment (onestacked) [reply bridgev2/bridgeconfig/legacymigrate.go]: Removed it.

**Comment by onestacked** [comment]
@tulir Can you approve the CI?

**Comment by onestacked** [comment]
@tulir Friendly ping :)

**Comment by jkolo** [comment]
https://github.com/element-hq/synapse/pull/17705 was merged into synapse develop :)

**Pull request closed without merging**